From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paolo Sabatino <paolo.sabatino@gmail.com>
Date: Mon, 6 May 2024 15:50:14 +0100
Subject: [ARCHEOLOGY] rockchip: add reset props for usb on rk322x

> X-Git-Archeology: - Revision 5657ec0798045ad9cff0df0033ff1c963dfcdd66: https://github.com/armbian/build/commit/5657ec0798045ad9cff0df0033ff1c963dfcdd66
> X-Git-Archeology:   Date: Mon, 06 May 2024 15:50:14 +0100
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: rockchip: add reset props for usb on rk322x
> X-Git-Archeology:
> X-Git-Archeology: - Revision 4b51f88e56c54e3b517e584916241c4942dbfc5f: https://github.com/armbian/build/commit/4b51f88e56c54e3b517e584916241c4942dbfc5f
> X-Git-Archeology:   Date: Sun, 02 Jun 2024 09:23:31 +0200
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: rk322x: keep usb resets deasserted on exit
> X-Git-Archeology:
> X-Git-Archeology: - Revision 7876017d0b77bbfefbb3d112045b32d9b50db928: https://github.com/armbian/build/commit/7876017d0b77bbfefbb3d112045b32d9b50db928
> X-Git-Archeology:   Date: Tue, 02 Jul 2024 23:31:50 +0000
> X-Git-Archeology:   From: Paolo <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: Bump rk322x-box and rk3318-box to u-boot v2024.07-rc5 (#6855)
> X-Git-Archeology:
---
 drivers/usb/host/dwc2.c                     | 5 ++++-
 drivers/usb/host/ehci-generic.c             | 6 +++---
 dts/upstream/src/arm64/rockchip/rk3328.dtsi | 8 ++++++++
 3 files changed, 15 insertions(+), 4 deletions(-)

diff --git a/drivers/usb/host/dwc2.c b/drivers/usb/host/dwc2.c
index 111111111111..222222222222 100644
--- a/drivers/usb/host/dwc2.c
+++ b/drivers/usb/host/dwc2.c
@@ -1438,7 +1438,10 @@ static int dwc2_usb_remove(struct udevice *dev)
 
 	dwc2_uninit_common(priv->regs);
 
-	reset_release_bulk(&priv->resets);
+	// Assert first and then leave the resets deasserted
+	reset_assert_bulk(&priv->resets);
+	reset_deassert_bulk(&priv->resets);
+
 	clk_disable_bulk(&priv->clks);
 	clk_release_bulk(&priv->clks);
 
diff --git a/drivers/usb/host/ehci-generic.c b/drivers/usb/host/ehci-generic.c
index 111111111111..222222222222 100644
--- a/drivers/usb/host/ehci-generic.c
+++ b/drivers/usb/host/ehci-generic.c
@@ -148,9 +148,9 @@ static int ehci_usb_remove(struct udevice *dev)
 	if (ret)
 		return ret;
 
-	ret = reset_release_bulk(&priv->resets);
-	if (ret)
-		return ret;
+	// Assert first and then leave the resets deasserted
+	reset_assert_bulk(&priv->resets);
+	reset_deassert_bulk(&priv->resets);
 
 	return clk_release_bulk(&priv->clocks);
 }
diff --git a/dts/upstream/src/arm64/rockchip/rk3328.dtsi b/dts/upstream/src/arm64/rockchip/rk3328.dtsi
index 111111111111..222222222222 100644
--- a/dts/upstream/src/arm64/rockchip/rk3328.dtsi
+++ b/dts/upstream/src/arm64/rockchip/rk3328.dtsi
@@ -977,6 +977,8 @@
 		g-tx-fifo-size = <256 128 128 64 32 16>;
 		phys = <&u2phy_otg>;
 		phy-names = "usb2-phy";
+		resets = <&cru SRST_USB2OTG>;
+		reset-names = "otg";
 		status = "disabled";
 	};
 
@@ -987,6 +989,8 @@
 		clocks = <&cru HCLK_HOST0>, <&u2phy>;
 		phys = <&u2phy_host>;
 		phy-names = "usb";
+		resets = <&cru SRST_USB2HOST_EHCIPHY>;
+		reset-names = "ehci";
 		status = "disabled";
 	};
 
@@ -997,6 +1001,8 @@
 		clocks = <&cru HCLK_HOST0>, <&u2phy>;
 		phys = <&u2phy_host>;
 		phy-names = "usb";
+		resets = <&cru SRST_USB2HOST_EHCIPHY>;
+		reset-names = "ehci";
 		status = "disabled";
 	};
 
@@ -1010,6 +1016,8 @@
 			      "bus_clk";
 		dr_mode = "otg";
 		phy_type = "utmi_wide";
+		resets = <&cru SRST_USB3OTG>;
+		reset-names = "otg";
 		snps,dis-del-phy-power-chg-quirk;
 		snps,dis_enblslpm_quirk;
 		snps,dis-tx-ipgap-linecheck-quirk;
-- 
Armbian

